src/status_im/extensions/ethereum.cljs

```diff
@@ -344,6 +344,17 @@
                                :method           constants/web3-personal-sign})}
                (navigation/navigate-to-cofx :wallet-sign-message-modal nil)))))
 
+(handlers/register-handler-fx
+ :extensions/ethereum-create-address
+ (fn [_ [_ _ {:keys [on-result]}]]
+   (let [args {:jsonrpc "2.0"
+               :method constants/status-create-address}
+         payload (types/clj->json args)]
+     (status/call-private-rpc payload #(let [{:keys [error result]} (types/json->clj %1)
+                                             response (merge {:result result} (when error {:error error}))]
```

Comment:
Instead of using merge (merge is generally used when the second argument is a symbol), you can make it probably clearer in a few ways:
```
response (cond-> {:result result}
                   error (assoc :error error))
```
or more declarative but with a bit of duplication:

```
response (if error 
                    {:result result :error error}
                    {:result result})
```
You could also just leave error as nil, if you know the function will check the content of the key and not the presence of the key itself, but riskier:

```
response {:result result
                 :error error}
```